src/Illuminate/Support/Collection.php

```diff
@@ -135,7 +135,7 @@ public function mode($key = null)
 
         $collection = isset($key) ? $this->pluck($key) : $this;
 
-        $counts = new self;
+        $counts = new static;
```

Comment:
Changed to be consistent with the 61 other calls to `new static` in this class, the only other instance of `new self` being the one in `toBase()`.